### DISCUSSION TITLE
ADF diagnostic error ERROR NC_EHDFERR Error at HDF5 layer

### DISCUSSION THREAD

**Discussion by AdrianaBosso** [NCAR/ADF#290] [General]
Hi everyone

I am trying to run the ADF diagnostic package in CASPER. ( git clone [GitHub - NCAR/ADF: A unified collection of python scripts used to generate standard plots from CAM outputs.](https://github.com/NCAR/ADF)),
After clone the package in my Work directory, load all the modules and npl environment modify the config_cam_baseline_example.yaml and [config_amwg_default_plots.yaml](https://github.com/NCAR/ADF/blob/main/config_amwg_default_plots.yaml).
I get an error but is something related to nco and hdf5 library I think:
ERROR NC_EHDFERR Error at HDF5 layer
HINT: NC_EHDFERR errors indicate that the HDF5-backend to netCDF is unable to perform the requested task. NCO can receive this devilishly inscrutable error for a variety of possible reasons including: 1) The run-time dynamic linker attempts to resolve calls from the netCDF library to the HDF library with an HDF5 libhdf5.a that is incompatible with the version used to build NCO and netCDF. 2) The file system does not allow the HDF5 flock() function, as of HDF5 1.10.x, to enable multiple processes to open the same file for reading, a feature known as SWMR (Single Write Multiple Read). The fix is to disable the HDF5 flock() by setting an environment variable thusly: "export HDF5_USE_FILE_LOCKING=FALSE". 3) An incorrect netCDF4 library implementation of a procedure (e.g., nc_rename_var()) in terms of HDF function calls (e.g., HDF5Lmove()) manifests an error or inconsistent state within the HDF5 layer. This often occurs during renaming operations ([Current issues renaming dimensions and coordinates · Issue #597 · Unidata/netcdf-c](https://github.com/Unidata/netcdf-c/issues/597)). 4) Attempting to compress or decompress a netCDF4 dataset with a non-standard (i.e., non-DEFLATE) filter when the requisite shared library to encode/decode that compression filter is not present in either the default location (/usr/local/hdf5/lib/plugin) or in the user-configurable location referred to by the HDF5_PLUGIN_PATH environment variable. One can determine if missing plugin libraries are the culprit by dumping the hidden attributes of the dataset with, e.g., ncks --hdn -m in.nc or ncdump -s -h in.nc. Any variables with (hidden) "_Filter" attributes require the corresponding shared libraries to be located in HDF5_PLUGIN_PATH. Some HDF5 implementations (at least MacOSX with MacPorts as of 20200907) may also require explicitly setting the plugin path in the environment, even for the default location! To test this, re-try your NCO command after doing this: "export HDF5_PLUGIN_PATH=/usr/local/hdf5/lib/plugin". 5) Bad vibes.
nco_err_exit(): ERROR Short NCO-generated message (usually name of function that triggered error): nco_enddef()
nco_err_exit(): ERROR Error code is -101. Translation into English with nc_strerror(-101) is "NetCDF: HDF error"
nco_err_exit(): ERROR NCO will now exit with system call exit(EXIT_FAILURE)

Could anyone tell me which is the problem??

Thanks in advances

Adriana

**Comment by nusbaume** [comment]
A similar discussion thread has started on the DiscussCESM forum here:

https://bb.cgd.ucar.edu/cesm/threads/adf-diagnostic-error-error-nc_ehdferr-error-at-hdf5-layer.9453/#post-54793

If that discussion results in a satisfactory answer then I will post it here as well.  Thanks!
